core/src/main/java/io/aiven/klaw/service/TopicSyncControllerService.java

```diff
@@ -873,27 +887,34 @@ private List<TopicConfig> getTopicsFromKafkaCluster(String env, String topicName
             .getClusters(KafkaClustersType.KAFKA, tenantId)
             .get(envSelected.getClusterId());
 
-    List<TopicConfig> topicsList =
+    LoadTopicsResponse loadTopicsResponse =
         clusterApiService.getAllTopics(
             kwClusters.getBootstrapServers(),
             kwClusters.getProtocol(),
             kwClusters.getClusterName() + kwClusters.getClusterId(),
             kwClusters.getKafkaFlavor(),
-            tenantId);
+            tenantId,
+            resetTopicsCache);
 
     topicCounter = 0;
 
-    List<TopicConfig> topicFilteredList = topicsList;
+    Set<TopicConfig> topicFilteredList = loadTopicsResponse.getTopicConfigSet();
     // Filter topics on topic name for search
 
-    if (topicNameSearch != null && topicNameSearch.length() > 0) {
+    if (topicNameSearch != null
+        && topicNameSearch.length() > 0
+        && !loadTopicsResponse.isLoadingInProgress()) {
       final String topicSearchFilter = topicNameSearch;
       topicFilteredList =
-          topicsList.stream()
+          loadTopicsResponse.getTopicConfigSet().stream()
               .filter(topic -> topic.getTopicName().contains(topicSearchFilter))
-              .collect(Collectors.toList());
+              .collect(Collectors.toSet());
```

Comment:
Ah I think this will fix one of the bugs that I have longed noticed and never gotten around to fix 😄